packages/w3up-client/test/legacy-compat.node.test.js

```diff
@@ -25,45 +29,77 @@ const createReceiptsServer = (agentStore) =>
 /** @type {Test.Suite} */
 export const testLegacyCompatibility = {
   uploadFile: Test.withContext({
-    'should upload a file to the service via legacy client': async (
+    'should upload a file to the service with legacy client': async (
+      assert,
+      context
+    ) =>
+      testUploadFile(assert, {
+        ...context,
+        legacy: { client: true, space: false },
+      }),
+    'should upload a file to the service with legacy space': async (
+      assert,
+      context
+    ) =>
+      testUploadFile(assert, {
+        ...context,
+        legacy: { client: false, space: true },
+      }),
+    'should upload a file to the service with legacy client and space': async (
       assert,
-      { connection, provisionsStorage, agentStore }
-    ) => {
-      const receiptsServer = createReceiptsServer(agentStore)
-      const receiptsEndpoint = await new Promise((resolve) => {
-        receiptsServer.listen(() => {
-          // @ts-expect-error
-          resolve(new URL(`http://127.0.0.1:${receiptsServer.address().port}`))
-        })
-      })
+      context
+    ) =>
+      testUploadFile(assert, {
+        ...context,
+        legacy: { client: true, space: true },
+      }),
+  }),
+}
```

Comment:
These tests ensure that we can upload with legacy space + legacy client, legacy space + new client and new space + legacy client.